staging/src/k8s.io/client-go/tools/portforward/portforward.go

```diff
@@ -349,6 +359,10 @@ func (pf *PortForwarder) handleConnection(conn net.Conn, port ForwardedPort) {
 	errorStream, err := pf.streamConn.CreateStream(headers)
 	if err != nil {
 		runtime.HandleError(fmt.Errorf("error creating error stream for port %d -> %d: %v", port.Local, port.Remote, err))
+		_ = pf.streamConn.Close()
+		if pf.streamFailChan != nil {
+			close(pf.streamFailChan)
```

Comment:
can two handleConnection calls run concurrently? if both fail to open a stream, will they try to close the channel concurrently? will that panic?

Reply:
the `pf.streamConn.Close()` should be sufficient to make the caller exit the loop (will hit the `<-pf.streamConn.CloseChan()` case)

I don't see why streamFailChan is needed

Reply:
That's a good catch, since it might be susceptible to one.